### DISCUSSION TITLE
Payment Notification

### DISCUSSION THREAD

**Discussion by jperez8** [creagia/laravel-redsys#31] [General]
Hi, I'm testing in my local environment payments with redsys, but either RedsysSuccessfulEvent or RedsysUnsuccessfulEvent are being triggered, i guess this is normal behaviour in local environment? And if it is how can i test this events in my local machine?
The hole flow is working correctly.

I'm using sail and i set `REDSYS_ENVIRONMENT=test`

<img width="713" alt="image" src="https://github.com/creagia/laravel-redsys/assets/48395571/8a6bb906-65c5-41a0-bb10-fc2c6230fd5f">
<img width="561" alt="image" src="https://github.com/creagia/laravel-redsys/assets/48395571/cf6595cb-4fcd-4ceb-a341-da82b3b79bac">

**Comment by dtorras** [comment]
Events should be fired in every environment (local, test and production). Can you confirm you are receiving the response notification from Redsys?

You can check this in the `redsys_notification_logs` table. If you can't find a new notification row after a successful payment, you should review if you are exposing your local environment correctly so Redsys can POST a response to your application (https://laravel.com/docs/10.x/sail#sharing-your-site)

**Comment by jperez8** [comment]
`redsys_notification_logs` is empty.
It is mandatory to share my site in order to receive the events?

**Comment by dtorras** [comment]
Otherwise, Redsys can't communicate with your application because your app is in your local environment. I created the [Local Gateway](https://github.com/creagia/laravel-redsys#local-gateway) to test locally without sharing the application but unfortunately, [Laravel Sail can't POST request](https://github.com/laravel/sail/issues/103) to the same application.

**Comment by dtorras** [comment]
You can read the original Redsys documentation about the redirection method. You'll see how the notification and redirection work: https://pagosonline.redsys.es/conexion-redireccion.html
